### DISCUSSION TITLE
[IntelliSense] Add setting for customizing character-count limit

### DISCUSSION THREAD

**Discussion by MichaelAllenWarner** [tailwindlabs/tailwindcss#9223] [Ideas]
The VS Code Tailwind IntelliSense extension places a character-count limit on input-strings. The limit seems to be ~1000 characters (which I'm guessing is related to the `1000` that appears a few times in [this file](https://github.com/tailwindlabs/tailwindcss-intellisense/blob/master/packages/tailwindcss-language-service/src/completionProvider.ts)). If a string exceeds that limit, then only classes _before_ the 1000th character will get the IntelliSense treatment.

It would be nice to have a setting for extending that limit. I sometimes exceed it in complex situations, especially when working with a traditional templating language rather than a JavaScript framework (it's easier in JavaScript to abstract things out into functions, say).

**Comment by MichaelAllenWarner** [comment]
Hi @bradlc — am I right that the 1000-character limit comes from the 3 instances of that number in [this file](https://github.com/tailwindlabs/tailwindcss-intellisense/blob/master/packages/tailwindcss-language-service/src/completionProvider.ts)? Also: would you consider merging a PR that either bumped up the 1000-character limit (to 1500 or 2000, say) or added an option for customizing the limit? If so, I'd likely try to find time to make that contribution. I'm guessing that not a lot of other folks are hitting the limit, but I occasionally am.

**Comment by jerimiahwelch** [comment]
+ 1 for this.

We run into this limitation when applying Tailwind intellisense to Config file classlists.

It works for the first part of the config object, but not the last part of the object

![Screenshot 2024-10-09 at 2 26 10 PM](https://github.com/user-attachments/assets/cb880289-308e-4875-b716-513b5dcd7557)
